explainaboard/analysis/feature_funcs.py

```diff
@@ -65,36 +85,44 @@ def accumulate_vocab_from_samples(
 
 
 def feat_freq_rank(
-    existing_features: dict,
-    statistics: Any,
-    text_from_sample: Callable,
-    tokenizer: Tokenizer,
-):
+    sys_info: SysOutputInfo,
```

Comment:
It looks `feat_freq_rank`, `feat_num_oov`, and `feat_length_freq` functions need to avoid `_get_tokens`. Arguments `sys_info`, `text`, `side` are used to only invoke `_get_tokens`. If `_get_tokens` is invoked before calling these functions, they could become much simpler.

Reply:
I agree. I think it might be a good idea to refactor this in a later PR, but maybe not here just to limit the scope of changes from getting larger.